pkg/resourcemanager/vm/reconcile.go

```diff
@@ -52,9 +52,14 @@ func (g *AzureVirtualMachineClient) Ensure(ctx context.Context, obj runtime.Obje
 	// to overcome the issue with the lack of idempotence of the Create call
 	item, err := g.GetVirtualMachine(ctx, resourceGroup, resourceName)
 	if err == nil {
-		instance.Status.Provisioned = true
-		instance.Status.Provisioning = false
-		instance.Status.Message = resourcemanager.SuccessMsg
+		if *item.ProvisioningState == "Succeeded" {
+			instance.Status.Provisioned = true
+			instance.Status.Provisioning = false
+			instance.Status.Message = resourcemanager.SuccessMsg
+		} else {
+			instance.Status.Provisioned = false
+			instance.Status.Provisioning = true
+		}
```

Comment:
@haagha -- I think the following lines should be included in the truly provisioned case?
```
		instance.Status.ResourceId = *item.ID
		return true, nil
```

And the other case should not return?

Reply:
populated this and more status fields for this